src/SwedbankPay.Sdk.Infrastructure/JsonSerialization/Converters/CustomRegionInfoConverter.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Globalization;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace SwedbankPay.Sdk.JsonSerialization.Converters
+{
+    public class CustomRegionInfoConverter : JsonConverter<RegionInfo>
+    {
+        public override RegionInfo Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if (reader.TokenType == JsonTokenType.PropertyName)
+            {
+                reader.GetString();
+                reader.Read();
+            }
+
+            if (reader.TokenType == JsonTokenType.StartObject)
+            {
+                if (reader.Read())
+                {
+                    var uri = Read(ref reader, typeToConvert, options);
+                    // Reads the EndObject
+                    reader.Read();
+                    return uri;
+                }
+            }
+
+            if (reader.TokenType == JsonTokenType.String)
+            {
+                var regionInfoString = reader.GetString();
+
+                if (string.IsNullOrEmpty(regionInfoString))
+                    return null;
+                return new RegionInfo(regionInfoString);
+            }
+
+            if (reader.TokenType == JsonTokenType.Null)

```

Comment:
Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-dotnet/pullRequest?prid=6290774)